server/config.js

```diff
@@ -12,22 +15,27 @@ export default function getConfig (dir) {
   return cache.get(dir)
 }
 
-async function loadConfig (dir) {
-  const path = join(dir, 'package.json')
-
-  let data
-  try {
-    data = await readFile(path, 'utf8')
-  } catch (err) {
-    if (err.code === 'ENOENT') {
-      data = '{}'
-    } else {
-      throw err
-    }
+function loadConfig (dir) {
+  const path = join(dir, 'next.config.js')
+  const packagePath = join(dir, 'package.json')
+
+  let userConfig = {}
+  let packageConfig = null
+
+  const userHasConfig = existsSync(path)
+  if (userHasConfig) {
+    const userConfigModule = require(path)
+    userConfig = userConfigModule.default || userConfigModule
   }
 
-  // no try-cache, it must be a valid json
-  const config = JSON.parse(data).next || {}
+  const userHasPackageConfig = existsSync(packagePath)
+  if (userHasPackageConfig) {
+    packageConfig = require(packagePath).next
+  }
+
+  if (packageConfig) {
+    console.warn("WARNING: You're using package.json as source of config for next.js. Please use next.config.js instead.")
```

Comment:
Change this to: 

console.log("> [warn] You are using package.json for next.js config. Use next.config.js instead.");

> Just FYI: Try not to use "Please" in front of messages. It's sounds like commanding. 
> I tend to use that and one of my UK friend asked to drop that. Messages sounds nice after that :)